### PR TITLE
Add support for vscode:extension/ resolver and http(s):// resolver

### PR DESCRIPTION
**Pull request by benoitf**
Allow to grab some vscode extension or download files hosted on a http(s) server

theia-vscode

### CODE REVIEW

**Review by evidolob — APPROVED** [review]

File: packages/plugin-ext-vscode/src/node/plugin-vscode-backend-module.ts [hunk 1]
```diff
@@ -6,14 +6,17 @@
...
 import { PluginVsCodeDirectoryHandler } from "./plugin-vscode-directory-handler";
 import { VsCodePluginScanner } from "./scanner-vscode";
+import { VsCodePluginDeployerResolver } from './plugin-vscode-resolver';
 
 export default new ContainerModule(bind => {
     bind(PluginDeployerFileHandler).to(PluginVsCodeFileHandler).inSingletonScope();
     bind(PluginDeployerDirectoryHandler).to(PluginVsCodeDirectoryHandler).inSingletonScope();
     bind(PluginScanner).to(VsCodePluginScanner).inSingletonScope();
+    bind(PluginDeployerResolver).to(VsCodePluginDeployerResolver).inSingletonScope();
+
```
Comment (mmorhun) [inline packages/plugin-ext-vscode/src/node/plugin-vscode-backend-module.ts]: Do we need this extra line?

File: packages/plugin-ext/src/main/node/plugin-http-resolver.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            const dirname = path.dirname(link.pathname);
+            const basename = path.basename(link.pathname);
+            console.log('basename is', basename);
+            console.log('dirname is', dirname);
+
+            const filename = dirname.replace(/\W/g, '_') + ('-') + basename;
+            console.log('filename is', filename);
+            console.log('dirname is', dirname);
+            /*
+                        let filename = parsed.pathname pluginResolverContext.getOriginId().replace(/\W/g, '_');
```
Comment (mmorhun) [inline packages/plugin-ext/src/main/node/plugin-http-resolver.ts]: Is it needed? If so, format it properly, please.

**Review by mmorhun — APPROVED** [review]

File: packages/plugin-ext/src/main/node/plugin-http-resolver.ts (hunk 2 above)
Comment (benoitf) [reply packages/plugin-ext/src/main/node/plugin-http-resolver.ts]: good catch :-) it shouldn't have reached the PR

File: packages/plugin-ext-vscode/src/node/plugin-vscode-backend-module.ts (hunk 1 above)
Comment (benoitf) [reply packages/plugin-ext-vscode/src/node/plugin-vscode-backend-module.ts]: fixed

File: packages/plugin-ext/src/main/node/plugin-http-resolver.ts (hunk 2 above)
Comment (benoitf) [reply packages/plugin-ext/src/main/node/plugin-http-resolver.ts]: fixed

**Review by olexii4 — APPROVED** [review]

**Comment by slemeur** [comment]
Impressive !

**Pull request merged**